File: regatta/eval/_comet_ramp.js
// THE COLOUR RAMP, BOTH CANDIDATES, OVER REAL VENUE WATER.
//
// `wind` is what ships; `heat` is the literal white -> green -> yellow -> orange -> red
// proposal, kept in STREAK_PALETTES for exactly this comparison. Draws each as swatches
// across the knot scale, over each venue's own water, because the two objections recorded
// against the literal ramp are both about what it sits ON: the fleet's orange at 26 kt, and
// Gatorgrass's olive water at 7-14 kt. A ramp cannot be judged on white.
const { chromium } = require('playwright');
const path = require('path');
const fs = require('fs');
const OUT = process.env.SHOT_OUT || '/private/tmp/claude-501/-Users-wesdyer-Desktop-wesdyer-github-io/0b98d4e5-b137-4a82-9d99-591fe88704f5/scratchpad';
(async () => {
  const b = await chromium.launch();
  const p = await b.newPage({ viewport: { width: 1200, height: 700 } });
  const errs = []; p.on('pageerror', e => errs.push(e.message));
  await p.goto('file://' + path.resolve('regatta/index.html'));
  await p.waitForTimeout(1300);
  const data = await p.evaluate(() => {
    const out = { palettes: {}, waters: {}, ktMax: STREAK_KT_MAX };
    for (const name of Object.keys(STREAK_PALETTES)) {
      const lut = buildStreakLut(name);
      out.palettes[name] = lut;
    }
    for (const v of ['lake', 'swamp', 'ocean', 'glowtide', 'redrock']) {
      const d = window.VENUE_DOC && window.VENUE_DOC[v];
      out.waters[v] = (d && d.palette && d.palette.baseColor) || '#0ea5e9';
    }
    return out;
  });
  fs.writeFileSync(`${OUT}/_ramp.json`, JSON.stringify(data));
  console.log('  palettes:', Object.keys(data.palettes).join(', '), ' scale 0 -', data.ktMax, 'kt');
  console.log('errors', errs.length ? errs.slice(0, 3) : 'none');
  await b.close();
})();

File: regatta/eval/_comet_ramp.py
"""Draw the two comet colour ramps over each venue's water. Run eval/_comet_ramp.js first."""
import json
from PIL import Image, ImageDraw

OUT = "/private/tmp/claude-501/-Users-wesdyer-Desktop-wesdyer-github-io/0b98d4e5-b137-4a82-9d99-591fe88704f5/scratchpad"
d = json.load(open(f"{OUT}/_ramp.json"))
waters, pals, ktmax = d["waters"], d["palettes"], d["ktMax"]
names = list(pals.keys())
W, ROW, PAD, LBL = 1100, 46, 10, 120
H = PAD + len(waters) * len(names) * (ROW + 6) + 40
img = Image.new("RGB", (W, H), (26, 26, 30))
dr = ImageDraw.Draw(img)
y = PAD
for vname, hexc in waters.items():
    wr = tuple(int(hexc.lstrip("#")[i:i+2], 16) for i in (0, 2, 4))
    for pname in names:
        lut = pals[pname]
        dr.rectangle([LBL, y, W - 10, y + ROW], fill=wr)
        n = len(lut)
        for i in range(n):
            x0 = LBL + (W - 10 - LBL) * i / n
            x1 = LBL + (W - 10 - LBL) * (i + 1) / n
            # A comet is a thin mark, so draw it as one — a full-height block flatters any ramp.
            dr.rectangle([x0, y + ROW * 0.38, x1, y + ROW * 0.62], fill=tuple(lut[i]))
        dr.text((6, y + ROW / 2 - 6), f"{vname[:8]} {pname}", fill=(230, 230, 235))
        y += ROW + 6
for k in range(0, ktmax + 1, 5):
    x = LBL + (W - 10 - LBL) * k / ktmax
    dr.line([x, y, x, y + 6], fill=(150, 150, 160))
    dr.text((x - 6, y + 8), f"{k}", fill=(190, 190, 200))
img.save(f"{OUT}/comet-ramps.png")
print("  wrote comet-ramps.png  (thin bar = the mark's actual weight; scale 0 ->", ktmax, "kt)")
